velox/exec/tests/utils/PlanBuilder.h

```diff
@@ -649,6 +649,49 @@ class PlanBuilder {
       const std::vector<std::string>& unnestColumns,
       const std::optional<std::string>& ordinalColumn = std::nullopt);
 
+  // Structure for a WindowFrame. It specifies :
+  // i)   The windowType (ROWS or RANGE)
+  // ii)  The starting and ending bounds PRECEDING (could be UNBOUNDED)
+  //      FOLLOWING (could be UNBOUNDED) or CURRENT_ROW
+  // iii) The start(or end) BoundString could be a constant or field reference.
+  struct WindowFrame {
```

Comment:
Can we use SQL string instead of this struct?

Reply:
I went a bit back and forth about this. There isn't an easy way to just parse the frame out from the DuckDB parser by itself. However, we could change the PlanBuilder::window API entirely. 

We could change the PlanBuilder window function to take a list of function sql strings like 
"window1(c) over (partition by a order by b rows between 10 preceding and current row) as d" and parse it into a WindowNode::function with a separate partition and order by clauses. 

All function invocations would require the same partition and order by clauses. wdyt ?

Reply:
It is unfortunate that we'll have to repeat `partition by a order by b` for all window functions, but I guess there is no way around that. Let's go with this approach.